src/main/java/programminglife/model/drawing/DrawableSNP.java

```diff
@@ -39,9 +39,10 @@
         this.child = (DrawableSegment) child;
         this.mutations = mutations.stream().map(DrawableSegment.class::cast).collect(Collectors.toSet());
 
-        this.parent.getChildren().clear();
+        this.mutations.stream().map(DrawableSegment::getIdentifier).forEach(this.parent.getChildren()::remove);
```

Comment:
I would say that it is better to create a method `removeChildren(Collection<Integer> childIds)` in the parent (i.e. DrawableNode, which would need to be need to be overriden)